internal/rest/avi_obj_vsvip.go

```diff
@@ -75,34 +75,25 @@ func (rest *RestOperations) AviVsVipBuild(vsvip_meta *nodes.AviVSVIPNode, vsCach
 		vsvip.DNSInfo = dns_info_arr
 		vsvip.VsvipCloudConfigCksum = &cksumstr
 
-		// handling static IP and networkName (infraSetting) updates.
-		vip := &avimodels.Vip{
-			VipID:                  &vipId,
-			AutoAllocateIP:         &autoAllocate,
-			AutoAllocateFloatingIP: vsvip_meta.EnablePublicIP,
-		}
-
-		// This would throw an error for advl4 the error is propagated to the gateway status.
-		if vsvip_meta.IPAddress != "" {
-			vip.IPAddress = &avimodels.IPAddr{Type: &ipType, Addr: &vsvip_meta.IPAddress}
-		}
-
 		if lib.IsPublicCloud() && lib.GetCloudType() != lib.CLOUD_GCP {
-			vips := networkNamesToVips(vsvip_meta.VipNetworks, vsvip_meta.EnablePublicIP)
-			vsvip.Vip = []*avimodels.Vip{}
-			vsvip.Vip = append(vsvip.Vip, vips...)
+			networkNamesToVips(vsvip, vsvip_meta.VipNetworks, vsvip_meta.EnablePublicIP)
 		} else {
 			// Set the IPAM network subnet for all clouds except AWS and Azure
 			if len(vsvip_meta.VipNetworks) != 0 {
-				if vip.IPAMNetworkSubnet == nil {
-					vip.IPAMNetworkSubnet = &avimodels.IPNetworkSubnet{}
+				if len(vsvip.Vip) == 1 {
+					vip := vsvip.Vip[0]
+					networkRef := "/api/network/?name=" + vsvip_meta.VipNetworks[0].NetworkName
+					vip.IPAMNetworkSubnet.NetworkRef = &networkRef
+					vsvip.Vip = []*avimodels.Vip{vip}
```

Comment:
This code implies that we will put whatever networkName is provided by the user into the vip object, keeping everything else including the subnet unchanged.
Can the user not associate a different subnet with the vip object along with a network change?

Reply:
If really this has been the behaviour then I am afraid the vip network switch just does not work. Since each vip network would also have an associated subnet. My suggestion would be to not allow this switch at all, keep the code simple and just call it out in the doc.

Reply:
one network to another would still work, 
same network different subnet won’t I believe
the current implementation is limited.

Reply:
But you are copying the rest of the vip object, which implies the subnet of the older network would be copied into the new network as well, is that practical?